### PR TITLE
en/ros2: add go-to setpoint type to interface lib documenation

### PR DESCRIPTION
**Pull request by tstastny**
Adds documentation of the go-to setpoint type. Would be nice to additionally add the screencast of what the example mode does, but I didnt know how to do that. @hamishwillee @bkueng see from the ros2-lib PR here: https://github.com/Auterion/px4-ros2-interface-lib/assets/8026163/76795ebc-13b9-4637-b7f7-867e773047fa

Needs to wait for
- [x] https://github.com/PX4/PX4-user_guide/pull/2740 
- [x] https://github.com/Auterion/px4-ros2-interface-lib/pull/4 
- [x] https://github.com/PX4/PX4-Autopilot/pull/22375

### CODE REVIEW

File: en/releases/main.md [hunk 1]
```diff
@@ -65,5 +65,5 @@ TBD ...
 ### ROS 2
 
 - [Experimental] [PX4 ROS 2 Interface Library](../ros2/px4_ros2_interface_lib.md): A new C++ library that simplifies controlling PX4 from ROS 2.
-  Supports adding flight modes in ROS 2 that are peers of the PX4 modes running on the flight controller.
-  Added to PX4 in [PX4-Autopilot#20707](https://github.com/PX4/PX4-Autopilot/pull/20707) (initial support).
+  Supports adding flight modes in ROS 2 that are peers of the PX4 modes running on the flight controller. 
+  Added to PX4 in [PX4-Autopilot#20707](https://github.com/PX4/PX4-Autopilot/pull/20707) (initial support).
```
Comment (hamishwillee) [inline en/releases/main.md:69]: @tstastny Add additional info about changed feature, including links to the PX4 pull requests associated with them.

**Comment by hamishwillee** [comment]
I've subedited to my taste and run prettier. This looks good. Of course we will add the video etc when presentation ready.

File: en/ros2/px4_ros2_interface_lib.md [hunk 2]
```diff
@@ -343,11 +344,56 @@ The other setpoint types are currently experimental, and can be found in: [px4_r
...
+```cpp
+const Eigen::Vector3f target_position_m{-10.F, 0.F, 3.F};
+const float heading_rad = 3.14F;
+_goto_setpoint->update(
+  target_position_m,
+  heading_rad);
+```
+
+An additional feature of the go-to setpoint is dynamic control on the underlying smoothers' speed limits (i.e. maximum horizontal and vertical translational velocities as well as heading rate).
+If, as above, left unspecified, the smoothers will default the the vehicle's default maximums (typically set to the physical limitations).
```
Comment (bkueng) [inline en/ros2/px4_ros2_interface_lib.md:375]:
```suggestion
If, as above, left unspecified, the smoothers will default to the vehicle's default maximums (typically set to the physical limitations).
```

File: en/ros2/px4_ros2_interface_lib.md [hunk 3]
```diff
@@ -335,6 +335,7 @@ The used types also define the compatibility with different vehicle types.
 
 The following sections provide a list of supported setpoint types:
 
+- [GotoSetpointType](#goto-setpoint-gotosetpointtype): Smooth position and (optionally) heading control
```
Comment (hamishwillee) [inline en/ros2/px4_ros2_interface_lib.md:338]:
```suggestion
- [GotoSetpointType](#go-to-setpoint-gotosetpointtype): Smooth position and (optionally) heading control
```

**Comment by bkueng** [comment]
https://github.com/Auterion/px4-ros2-interface-lib/pull/4 got merged and I updated the PR. @hamishwillee good to merge?

**Comment by hamishwillee** [comment]
Yes. Thank you.

**Pull request merged**